opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts

```diff
@@ -74,39 +62,50 @@ export const getOwnerId = (workspace: BasicStoreEntityWorkspace) => {
   return (Array.isArray(workspace.creator_id) && workspace.creator_id.length > 0) ? workspace.creator_id.at(0) : workspace.creator_id;
 };
 
-export const objects = async (context: AuthContext, user: AuthUser, workspaceId: string, args: WorkspaceObjectsArgs) => {
-  const key = buildRefRelationKey(RELATION_HAS_REFERENCE);
-  const types = args.types ?? ['Stix-Meta-Object', 'Stix-Core-Object', 'stix-relationship'];
-  const filters = [{ key, values: [workspaceId] }, ...(args.filters || [])];
+export const objects = async (context: AuthContext, user: AuthUser, { investigated_entities_ids }: BasicStoreEntityWorkspace, args: WorkspaceObjectsArgs) => {
+  const types = args.types ?? INVESTIGABLE_TYPES;
+
+  if (!investigated_entities_ids) {
+    return { edges: [] };
+  }
+
+  const filters = [
+    { key: 'internal_id', values: investigated_entities_ids },
+    ...(args.filters ?? [])
+  ];
   const finalArgs = { ...args, filters };
   if (args.all) {
     return paginateAllThings(context, user, types, finalArgs);
   }
   return listThings(context, user, types, finalArgs);
 };
 
-export const toStixReportBundle = async (context: AuthContext, user: AuthUser, workspace: BasicStoreEntityWorkspace): Promise<string> => {
-  if (workspace.type !== 'investigation') {
-    throw FunctionalError('You can only export investigation objects as a stix report bundle');
+const isEntityInvestigable = (entity: BasicStoreEntity): boolean => {
+  const matching_types = INVESTIGABLE_TYPES
+    .filter((investigable_type) => entity.parent_types.includes(investigable_type));
+
+  return matching_types.length !== 0;
+};
+
+const checkEntitiesAreInvestigable = (entities: Array<BasicStoreEntity>): void => {
+  entities
+    .filter((entity) => !isEntityInvestigable(entity))
+    .forEach((entity) => { throw FunctionalError(`Entity with id '${entity.id}' of type '${entity.entity_type}' is not investigable.`); });
+};
+
+const checkMissingEntities = (AddedOrReplacedinvestigatedEntitiesIds: (string | null)[], entities: Array<BasicStoreEntity>): void => {
```

Comment:
In which case we can have that ?

Reply:
When doing a `fieldPatch` on `investigated_entities_ids`. We want to make sure entities ids passed exist in the DB.